rdrf/rdrf/email_notification.py

```diff
@@ -17,55 +16,111 @@ class RdrfEmail(object):
 
     _DEFAULT_LANGUAGE = "en"
 
-    def __init__(self, reg_code=None, description=None, language="en", email_notification=None):
+    def __init__(self, reg_code=None, description=None, email_notification=None, language=None):
         self.email_from = None
-        self.recipient = []
+        self.recipients = []
         self.email_templates = []
-        self.language = language
         self.template_data = {}
 
         self.reg_code = reg_code
         self.description = description
+        self.language = language # used to only send to subset of languages by EmailNotificationHistory resend
 
-        self.email_notification = email_notification
+        if email_notification:
+            self.email_notification = email_notification
+            self.reg_code = self.email_notification.registry.code
+            self.description = self.email_notification.description
+            
+        else:
+            self.email_notification = self._get_email_notification()
+            
 
     def send(self):
```

Comment:
The code before this change had one recipients list built up from the recipient field and the group recipient field and one send ( to those recipients) - with the language being determined from the request )  - I'll take a look again --  
there is the resend functionalitty via the email notification history to take into account as well - ( I did actually refactor this  to make it work) - the original resend was at the level of the one email notifcation ( ie all recipients would receive the communication (again))  - currently it's at the language level( I didn't have to change the ENH listing view at all for this) if we make it at the level of user , we might have to change the gui further